### PR TITLE
heketi-cli: heketi-cli command to return help

### PR DESCRIPTION
**Pull request by MohamedAshiqrh**
Signed-off-by: Mohamed Ashiq Liyazudeen <mliyazud@redhat.com>

### CODE REVIEW

**Comment by lpabon** [comment]
@MohamedAshiqrh could you describe this a little more, I am not sure what this is fixing.

**Comment by lpabon** [comment]
test this please

**Comment by MohamedAshiqrh** [comment]
# heketi-cli                                  
Usage:
  heketi-cli [flags]
  heketi-cli [command]

Examples:
  $ export HEKETI_CLI_SERVER=http://localhost:8080
  $ heketi-cli volume list

Available Commands:
  cluster                        Heketi cluster management
  device                         Heketi device management
  setup-openshift-heketi-storage Setup OpenShift/Kubernetes persistent storage for Heketi
  node                           Heketi Node Management
  topology                       Heketi Topology Management
  volume                         Heketi Volume Management

Flags:
  -h, --help[=false]: help for heketi-cli
      --json[=false]: 
	Print response as JSON
      --log-flush-frequency=5s: Maximum number of seconds between log flushes
      --secret="": 
	Secret key for specified user.  Can also be
	set using the environment variable HEKETI_CLI_KEY
  -s, --server="": 
	Heketi server. Can also be set using the
	environment variable HEKETI_CLI_SERVER
      --user="": 
	Heketi user.  Can also be set using the
	environment variable HEKETI_CLI_USER
  -v, --version[=false]: 
	Print version

Use "heketi-cli [command] --help" for more information about a command.

**Comment by lpabon** [comment]
@MohamedAshiqrh awesome

**Comment by lpabon** [comment]
retest this please

**Comment by lpabon** [comment]
@MohamedAshiqrh FYI, the command "test this please" and "retest this please" are not for you, they are commands for the CI system :smile: 

https://ci.centos.org/view/Gluster/job/gluster_heketi-functional/

**Comment by lpabon** [comment]
Maybe I should do:

@centos-ci retest this please

That's probably better ;-)

File: client/cli/go/cmds/crawl.xml [hunk 1]
```diff
@@ -0,0 +1 @@
+<url>"tool"</url>
```
Comment (lpabon) [inline client/cli/go/cmds/crawl.xml]: Is this file needed?

**Comment by lpabon** [comment]
This is not working for me:

```
(heketi) lpabon@localhost : ~/git/golang/heketi/src/github.com/heketi/heketi/client/cli/go
$ ./heketi-cli 
Server must be provided


(heketi) lpabon@localhost : ~/git/golang/heketi/src/github.com/heketi/heketi/client/cli/go
$ ./heketi-cli -v
heketi-cli v2.0.3-dev-2-ge414e7a-pr_410

$ git log --oneline | head
e414e7a heketi-cli: heketi-cli command to return help

```

File: client/cli/go/cmds/crawl.xml (hunk 1 above)
Comment (MohamedAshiqrh) [inline client/cli/go/cmds/crawl.xml]: No this is not needed.

**Comment by MohamedAshiqrh** [comment]
I had server already set, I think this patch needs little more changes to make help work without server set.

**Comment by lpabon** [comment]
Without the server set, now I get this error:

```
$ ./heketi-cli cluster list
Error: Get /clusters: unsupported protocol scheme ""
```

**Comment by lpabon** [comment]
When changing the behavior, please check that all other calls are unaffected.

**Comment by obnoxxx** [comment]
I think that disabling the "Server must be provided" message is a different issue.
I agree that it does not make a lot of sense to have this message instead of the usage when just the naked heketi-cli cmd is called. But currently having the server set is a prerequisite for further processing, so imho a fix for this could be valid if it works with this assumption.

**Comment by MohamedAshiqrh** [comment]
@obnoxxx yeah, I am not sure about the way to fix this. But in search of the fix have learned that there can be a lot of improvements in cli. IMO, The easy-fix can be changing the version from flag to command. only operation that heketi-cli command does is prints the version, If we remove that from heketi-cli command. Then it will print usage and version will be a command instead of flag.
`# heketi-cli --version` to `# heketi-cli version`

**Comment by MohamedAshiqrh** [comment]
Hey,

So the reason why the help is not being printed on 'heketi-cli' invoke is because do version handling. Was looking into kubectl code, They have really cool cli with utils, templates, etc. which can implemented in future. For our current problem, how they handle is they write a bash function which will be invoked on setting a flag. I am trying this [one](https://godoc.org/github.com/spf13/cobra#Command.MarkFlagCustom) will let know how it goes.

https://github.com/heketi/heketi/pull/475

**Comment by obnoxxx** [comment]
@MohamedAshiqrh Ok. I see now what you mean with the version. I think it is not necessary to change --version to a command in order to have the root cmd print usage. I have set up a PR #477 which containes a modification of your earlier patch which works for me. The point is that your check for RootCmd.Flags().Args() does not seem to have done the trick. I check os.Args[1:] instead, which may not be the best or correct solution, but works for me.

**Comment by lpabon** [comment]
I will review #477

**Comment by obnoxxx** [comment]
Closes #415

**Comment by obnoxxx** [comment]
But superseded by #477

**Comment by lpabon** [comment]
Closed by #477
